### PR TITLE
Map floatingbutton

### PR DESCRIPTION
**Pull request by jasontsh**
#314 #337 #338

### CODE REVIEW

File: PennMobile/src/main/java/com/pennapps/labs/pennmobile/MapFragment.java [hunk 1]
```diff
@@ -215,50 +431,351 @@ public boolean onQueryTextChange(String arg0) {
...
+            public boolean onClose() {
+                suggestionList.setVisibility(View.GONE);
+                return false;
+            }
+        };
+        searchView.setOnCloseListener(closeListener);
+    }
+
+    private void showSuggestion() {
+        final ArrayList<String> list = new ArrayList<>(5);
```
Comment (dlakata) [inline PennMobile/src/main/java/com/pennapps/labs/pennmobile/MapFragment.java]: should this MAX_SUGGESTION_SIZE?

Comment (jasontsh) [inline PennMobile/src/main/java/com/pennapps/labs/pennmobile/MapFragment.java]: Good catch.

**Comment by dlakata** [comment]
refactor findBusRoute, otherwise I suppose LGTM

**Comment by adelq** [comment]
Merge conflict

**Comment by adelq** [comment]
LGTM, needs to be manually merged due to merge conflicts.

**Pull request merged**